### DISCUSSION TITLE
Perform an API-side action on behalf of the user

### DISCUSSION THREAD

**Discussion by Etheonor** [supabase/supabase#761] [Questions]
Hello everyone!

I'm sure it's easy but my brain is really tired right now :p 

I want to update a row, and this action is limited for the users, only the owner of the row can update it. It's easy to do client side, but what if I want to make an API call to do that? Is there a way to add the user access token in the request? I know in the header of the request we have something like Authorization: Bearer + Api key, but if I also need the user JWT?

I'm not even sure if my question make sense!

**Comment by pmauchle** [comment]
You can send the access token to the API. Example from [nextjs-with-supabase-auth/pages/index.js](https://github.com/supabase/supabase/blob/af99c593bc2a86368d395173beaf51afe4600b05/examples/nextjs-with-supabase-auth/pages/index.js)

`const { data, error } = useSWR(session ? ['/api/getUser', session.access_token] : null, fetcher)`

On the API side, validate the token:
[nextjs-with-supabase-auth/pages/api/getUser.js](https://github.com/supabase/supabase/blob/master/examples/nextjs-with-supabase-auth/pages/api/getUser.js)

`const token = req.headers.token;
  const { data: user, error } = await supabase.auth.api.getUser(token);`

**Comment by eifo** [comment]
The token does not persist when using api routes, so get token
token = req.headers.token and then

supabase.auth.session = () => ({

  access_token: token,

 })

And then run await supabase query

Better off to include it in middleware and wrap every route with it.
